frontend/src/pages/register/form.tsx

```diff
@@ -1,76 +1,130 @@
-import React, { useState } from 'react'
+import React, { useState, useRef } from 'react'
+import { useForm, FormProvider, useFormContext } from 'react-hook-form'
 import { Form } from '../../components'
+import {
+  Error,
+  Input,
+  InputPassword,
+  PasswordContainer,
+  SeenIcon,
+} from '../../components/form/form-styles'
 import { Title } from '../../components/text/text-styles'
 
-const RegisterForm = () => {
-  const [email, setEmail] = useState<string>('')
-  const [password, setPassword] = useState<string>('')
-  const [confirmPassword, setConfirmPassword] = useState<string>('')
-  const [firstName, setFirstName] = useState<string>('')
-  const [lastName, setLastName] = useState<string>('')
+type Props = {
+  email: string
+  password: string
+  confirmPassword: string
+  firstName: string
+  lastName: string
+}
 
-  const emailChangeHandler = (event: React.FormEvent<HTMLInputElement>) => {
-    setEmail(event.currentTarget.value)
-    console.log(event.currentTarget.value)
-  }
+const RegisterForm = ({ loading, setFormData }: any) => {
```

Comment:
Avoid using `any`... Let's create some lint rule that won't allow the usage of any

`
  interface RegisterProps {
     isLoading: boolean;
     onSubmit: (formData:  FormData) => void
  }
`